tempesta_fw/t/unit/tfw_fuzzer.c

```diff
@@ -0,0 +1,563 @@
+#include <linux/kernel.h>
+#include <linux/string.h>
+#include <linux/slab.h>
+
+#include "tfw_fuzzer.h"
+
+struct fuzz_msg {
+	char *s;
+	int inval; /* 0 - valid, 1 - invalid */
+};
+
+static struct fuzz_msg methods[] = {{"GET", 0}, {"HEAD", 0}, {"POST", 0},
+	{NULL, }};
+static struct fuzz_msg uri_path_start[] = {{"http:/", 0}, {"https:/", 0},
+	{"", 0}, {NULL, }};
+static struct fuzz_msg uri_file[] = {{"file.html", 0}, {"f-i_l.e", 0},
+	{"fi%20le", 0}, {"xn--80aaxtnfh0b", 0}, {NULL, }};
+static struct fuzz_msg versions[] = {{"1.0", 0}, {"1.1", 0}, {"0.9", 1},
+	{NULL, }}; // HTTP/0.9 is blocked
+static struct fuzz_msg resp_code[] = {{"100 Continue", 0}, {"200 OK", 0},
+	{"302 Found", 0}, {"304 Not Modified", 0}, {"400 Bad Request", 0},
+	{"403 Forbidden", 0}, {"404 Not Found", 0},
+	{"500 Internal Server Error", 0}, {NULL, }};
+static struct fuzz_msg conn_val[] = {{"keep-alive", 0}, {"close", 0},
+	{"upgrade", 0}, {NULL, }};
+static struct fuzz_msg ua_val[] = {{"Wget/1.13.4 (linux-gnu)", 0},
+	{"Mozilla/5.0", 0}, {NULL, }};
+static struct fuzz_msg hosts[] = {{"localhost", 0}, {"127.0.0.1", 0},
+	{"example.com", 0}, {"xn--80aacbuczbw9a6a.xn--p1ai", 0}, {NULL, }};
+static struct fuzz_msg content_type[] = {{"text/html;charset=utf-8", 0},
+	{"image/jpeg", 0}, {"text/plain", 0}, {NULL, }};
+static struct fuzz_msg content_length[] = {{"0", 0}, {"10", 1}, {"-42", 1},
+	{"146", 1}, {"0100", 1}, {"100500", 1}, {NULL, }};
```

Comment:
It's good to test invalid string values like ```efg``` and very large numbers (large than 64 bit) like "289938893745782652477004635711254524". The same for chunks length in chunked transfer encoding.